python/sbp/acquisition.py

```diff
@@ -87,35 +90,51 @@ class AcqSvProfile(object):
                'cf',
                'cp',
               ]
-
-  def __init__(self, payload=None, **kwargs):
-    if payload:
-      self.from_binary(payload)
-    else:
-      self.job_type = kwargs.pop('job_type')
-      self.status = kwargs.pop('status')
-      self.cn0 = kwargs.pop('cn0')
-      self.int_time = kwargs.pop('int_time')
-      self.sid = kwargs.pop('sid')
-      self.bin_width = kwargs.pop('bin_width')
-      self.timestamp = kwargs.pop('timestamp')
-      self.time_spent = kwargs.pop('time_spent')
-      self.cf_min = kwargs.pop('cf_min')
-      self.cf_max = kwargs.pop('cf_max')
-      self.cf = kwargs.pop('cf')
-      self.cp = kwargs.pop('cp')
+  __zips__ = [
+              ('u8', 'job_type'),
+              ('u8', 'status'),
+              ('u16', 'cn0'),
+              ('u8', 'int_time'),
+              ('GnssSignal', 'sid'),
+              ('u16', 'bin_width'),
+              ('u32', 'timestamp'),
+              ('u32', 'time_spent'),
+              ('s32', 'cf_min'),
+              ('s32', 'cf_max'),
+              ('s32', 'cf'),
+              ('u32', 'cp'),
+             ]
 
   def __repr__(self):
     return fmt_repr(self)
+
+  def __getitem__(self, item):
+    return getattr(self, item)
+
   
-  def from_binary(self, d):
-    p = AcqSvProfile._parser.parse(d)
-    for n in self.__class__.__slots__:
-      setattr(self, n, getattr(p, n))
+  def from_binary(self, d, offset=0):
+    try:
+      size = 0
+      for t, s in AcqSvProfile.__zips__:
+        if t in TYPES_KEYS_NP:
+          a = np.ndarray(1, TYPES_NP[t], d, size + offset)
+          size += a.itemsize
+          setattr(self, s, a.item())
+        else:
+          o = globals()[t]()
+          size += o.from_binary(d, size + offset)
+          setattr(self, s, o)
+      return size
+    except:
```

Comment:
These try-catch blocks were mostly for debugging purposes during development. I'll remove them.